src/wizards/newspack/views/settings/additional-brands/brand-upsert.tsx

```diff
@@ -0,0 +1,409 @@
+/**
+ * Additional Brands Brand page. Used to edit and create.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs, cleanForSlug } from '@wordpress/url';
+import { Fragment, useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	Router,
+	Card,
+	Grid,
+	Button,
+	SectionHeader,
+	TextControl,
+	ImageUpload,
+	ColorPicker,
+	SelectControl,
+	RadioControl,
+	hooks,
+} from '../../../../../components/src';
+
+import './style.scss';
+
+const { useParams } = Router;
+
+const {
+	themeColors: registeredThemeColors,
+	menuLocations,
+	menus: availableMenus,
+} = window.newspackSettings[ 'additional-brands' ].sections.additionalBrands;
+
+export default function Brand( {
+	brands = [],
+	upsertBrand,
+	wizardApiFetch,
+	fetchLogoAttachment,
+}: {
+	brands: Brand[];
+	editBrand?: number;
+	upsertBrand: ( brandId: number, brand: Brand ) => void;
+	wizardApiFetch: WizardApiFetch;
+	fetchLogoAttachment: ( brandId: number, logoId: number ) => void;
+} ) {
+	const { brandId = '0' } = useParams();
+	const selectedBrand = brands.find( ( { id } ) => id === Number( brandId ) );
+
+	const [ brand, updateBrand ] = hooks.useObjectState< Brand >( {
+		id: 0,
+		name: 'Brand Name',
```

Comment:
Perhaps it might also be a good idea to have this as a placeholder instead of default input value.

Reply:
Updated 75833a3372b61af7bf1d0ae927aabc0acc145db0